### PR TITLE
gRPC client, server, and servlet initial support

### PR DESCRIPTION
**Pull request by wtlucy**
The work here replaces what was done in #9989.  This PR adds three new features, described below.  For all features, Liberty provides the required gRPC libraries, however apps must bundle their protos, stubs, and service implementations.  

1. `grpcServer-1.0` enables support for webapp-bundled gRPC services.  This feature works by scanning for classes that implement `io.grpc.BindableService` and registering them with a Liberty-managed gRPC server that runs directly on Liberty's HTTP/2 channel.  

2. `grpcServlet-1.0` also enables support for webapp-bundled gRPC services, however its implementation relies on Liberty's Servlet 4.0 functionality.  This feature works by using an asynchronous Servlet as the transport, whereas the `grpcServer` feature uses Liberty's channel framework directly. 

3. `grpcClient-1.0` provides apps with the Netty gRPC client.  At this stage, applications must directly create and manage their `ManagedChannel`s.  
 
For #10088 and #10089

### CODE REVIEW

File: dev/com.ibm.ws.grpc.client/.project [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+	<projects>
+	</projects>
+	<buildSpec>
+		<buildCommand>
+			<name>org.eclipse.jdt.core.javabuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+	</buildSpec>
+	<natures>
```
Comment (pnicolucci) [inline dev/com.ibm.ws.grpc.client/.project]: I think you'll want to / need to add the bndNature/bndBuilder here

File: dev/com.ibm.ws.grpc.client/bnd.bnd [hunk 2]
```diff
@@ -0,0 +1,64 @@
+#*******************************************************************************
+# Copyright (c) 2019 IBM Corporation and others.
```
Comment (pnicolucci) [inline dev/com.ibm.ws.grpc.client/bnd.bnd]: Should we update this to 2020?

File: dev/com.ibm.ws.grpc.client/.classpath [hunk 3]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<classpath>
+	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8"/>
```
Comment (pnicolucci) [inline dev/com.ibm.ws.grpc.client/.classpath]: bnd requires java11 should be use that here as well?

File: dev/com.ibm.ws.grpc.server/src/com/ibm/ws/grpc/GrpcApplicationManager.java [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+			LibertyServerBuilder serverBuilder = new LibertyServerBuilder();
+			for (BindableService service : services) {
+				serverBuilder.addService(service.bindService());
+			}
+			Server server = serverBuilder.build();
+			try {
+				server.start();
+				ActiveGrpcServers.addServer(appInfo.getName(), server);
+			} catch (IOException e) {
+				logMessage(Level.FINE, "initGrpcServices",
```
Comment (pnicolucci) [inline dev/com.ibm.ws.grpc.server/src/com/ibm/ws/grpc/GrpcApplicationManager.java]: Do we want to do more than log the message?

File: dev/com.ibm.ws.grpc.server/src/com/ibm/ws/grpc/GrpcApplicationManager.java [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+
+	@Override
+	public void applicationStopping(ApplicationInfo appInfo) {
+		Collection<Server> servers = ActiveGrpcServers.getServerList();
+		for (Server s : servers) {
+			System.out.println("applicationStopping removing grpc server: " + s);
+			s.shutdown();
+		}
+		ActiveGrpcServers.removeServer(appInfo.getName());
+		System.out.println("applicationStopping app: " + appInfo.getName());
```
Comment (pnicolucci) [inline dev/com.ibm.ws.grpc.server/src/com/ibm/ws/grpc/GrpcApplicationManager.java]: Just pointing out the System.out

File: dev/com.ibm.ws.grpc.server/bnd.bnd [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+  com.ibm.websphere.javaee.servlet.4.0;version=latest,\
+  com.ibm.ws.grpc.common.1.0;version=latest,\
+  com.google.code.findbugs:jsr305;version=3.0.2,\
+  com.google.guava:guava;version=27.0.1, \
+  com.ibm.ws.channelfw;version=latest,\
+  com.ibm.ws.transport.http;version=latest,\
+  com.ibm.ws.kernel.service
+  
+javac.source: 1.8
+javac.target: 1.8
```
Comment (pnicolucci) [inline dev/com.ibm.ws.grpc.server/bnd.bnd]: Does this make sense given the java11 requirement?

**Comment by wtlucy** [comment]
#build

File: dev/com.ibm.ws.grpc.server/bnd.bnd (hunk 6 above)
Comment (wtlucy) [reply dev/com.ibm.ws.grpc.server/bnd.bnd]: We only need java 8 at build time, so this does make sense

File: dev/com.ibm.ws.grpc.server/src/com/ibm/ws/grpc/GrpcApplicationManager.java (hunk 4 above)
Comment (wtlucy) [reply dev/com.ibm.ws.grpc.server/src/com/ibm/ws/grpc/GrpcApplicationManager.java]: I've added a FFDC for now.  We'll go back over messages more carefully in another PR.

File: dev/com.ibm.ws.grpc.client/.classpath (hunk 3 above)
Comment (wtlucy) [reply dev/com.ibm.ws.grpc.client/.classpath]: No other projects are doing this, so let's hold off for now.

**Comment by wtlucy** [comment]
#build

**Comment by wtlucy** [comment]
#build

**Comment by wtlucy** [comment]
#build

**Comment by wtlucy** [comment]
#build

**Comment by wtlucy** [comment]
#build

**Comment by wtlucy** [comment]
#run-libby-bot

**Pull request merged**